xbmc/filesystem/CurlFile.cpp

```diff
@@ -53,6 +53,11 @@ curl_proxytype proxyType2CUrlProxyType[] = {
 #define FILLBUFFER_NO_DATA    1
 #define FILLBUFFER_FAIL       2
 
+#if LIBCURL_VERSION_NUM < 0x073700 // 0.7.55
+#define CURLINFO_CONTENT_LENGTH_DOWNLOAD CURLINFO_CONTENT_LENGTH_DOWNLOAD_T
+#define CURLINFO_SPEED_DOWNLOAD CURLINFO_SPEED_DOWNLOAD_T
+#endif
+
```

Comment:
I don't think simply redefining will work because instead of a `double` a `curl_off_t` (which is `long` on my platform) is expected for the `_T` functions. The correct thing to do is probably to handle the cases at their calling side, e.g.:
```diff
diff --git a/xbmc/filesystem/CurlFile.cpp b/xbmc/filesystem/CurlFile.cpp
index 11a527211b..0448cba6a0 100644
--- a/xbmc/filesystem/CurlFile.cpp
+++ b/xbmc/filesystem/CurlFile.cpp
@@ -365,8 +365,14 @@ long CCurlFile::CReadState::Connect(unsigned int size)
       return -1;
   }
 
+#if LIBCURL_VERSION_NUM < 0x073700 // 0.7.55
   double length;
-  if (CURLE_OK == g_curlInterface.easy_getinfo(m_easyHandle, CURLINFO_CONTENT_LENGTH_DOWNLOAD, &length))
+  CURLcode result = g_curlInterface.easy_getinfo(m_easyHandle, CURLINFO_CONTENT_LENGTH_DOWNLOAD, &length);
+#else
+  curl_off_t length;
+  CURLcode result = g_curlInterface.easy_getinfo(m_easyHandle, CURLINFO_CONTENT_LENGTH_DOWNLOAD_T, &length);
+#endif
+  if (CURLE_OK == result)
   {
     if (length < 0)
       length = 0.0;
```

Reply:
You are right, I've fixed it.